librad/src/git/storage.rs

```diff
@@ -522,94 +619,206 @@ impl Storage<WithSigner> {
+                        );
+                        let certifier_id = Reference::rad_id(certifier_hash);
+                        self.backend
+                            .reference_symbolic(
+                                &certifier_here.to_string(),
+                                &certifier_id.to_string(),
+                                /* force */ false,
+                                &format!(
+                                    "Symref certifier: `{}` -> `{}`",
+                                    certifier_here, certifier_id
+                                ),
+                            )
+                            .and(Ok(()))
+                    },
+                }
+            })?;
         }
 
         // At this point, the transitive tracking graph may have changed. Let's
         // update the refs, but don't recurse here for now (we could, if
-        // we reload `self.refs()` and compare to the value we had
+        // we reload `self.rad_refs()` and compare to the value we had
         // before fetching).
-        self.update_refs(urn)
+        self.update_refs(&urn)
+    }
+
+    // DO NOT MAKE THIS PUBLIC YET
+    fn delete_repo(&self, urn: &RadUrn) -> Result<(), Error> {
+        References::from_globs(&self.backend, &[format!("refs/namespaces/{}/*", urn.id)])?
+            .try_for_each(|reference| reference?.delete())
+            .map_err(Error::from)
+    }
+
+    /// Persist [`User`] `id` as the default `rad/self` identity
+    pub fn set_default_rad_self(&self, id: User<Verified>) -> Result<(), Error> {
+        let urn = id.urn();
+        if !self.has_urn(&urn)? {
+            return Err(Error::NoSuchUrn(urn));
+        }
+
+        Config::try_from(&self.backend)?
+            .set_user(Some(id))
+            .map_err(Error::from)
+    }
+
+    /// Set the `rad/self` identity for `urn`
+    pub fn set_rad_self(&self, urn: &RadUrn, spec: RadSelfSpec) -> Result<(), Error> {
+        match spec {
+            RadSelfSpec::None => {
+                let have = Reference::rad_self(urn.id.clone(), None).find(&self.backend);
+                match have {
+                    Err(_) => Ok(()),
+                    Ok(mut reference) => reference.delete().map_err(Error::from),
+                }
+            },
+
+            some => {
+                let src = Reference::rad_self(urn.id.clone(), None);
+                let target = match some {
+                    RadSelfSpec::Default => {
+                        let id = self.default_rad_self()?;
+                        Ok::<_, Error>(Reference::rad_id(id.urn().id))
+                    },
+
+                    RadSelfSpec::Urn(self_urn) => {
+                        let meta: User<Draft> = self.metadata(&self_urn)?;
+                        Config::try_from(&self.backend)?.guard_user_valid(&meta)?;
+                        Ok(Reference::rad_id(self_urn.id))
+                    },
+
+                    RadSelfSpec::None => unreachable!(),
```

Comment:
I think it's possible to avoid this if you use `return` in the `None` case, and then do `let target = match spec { /* etc. */ }` followed by the `self.backend` code. It'd only be an issue if the first case was moved below, so no big deal :shrug:

Reply:
`return` what? The result of the pattern match is bound to a variable

Reply:
So I meant here: https://github.com/radicle-dev/radicle-link/pull/181/files#diff-b55217e1bbc5075454fe7f8b69de56fcR789-R790 you could call `return `Ok(())` and `return reference.delete().map_err(Error::from)`. Then float the `let target` as well of the rest of logic outside the outer `match`. I believe the `return` will still work for the type checker, as long as the rest of the arms of the `match` have the same types.

That way we'll have exhaustive pattern matching, and no `unreachable`.

Reply:
I don't get it, but may you like this better 7deb942

Reply:
:kissing_closed_eyes::ok_hand: